mailarchive/mailarchive-tool/tool/src/java/org/sakaiproject/mailarchive/entityprovider/MailArchiveEntityProvider.java

```diff
@@ -0,0 +1,822 @@
+import org.sakaiproject.entitybroker.entityprovider.EntityProvider;
+import org.sakaiproject.entitybroker.entityprovider.annotations.EntityCustomAction;
+import org.sakaiproject.entitybroker.entityprovider.capabilities.ActionsExecutable;
+import org.sakaiproject.entitybroker.entityprovider.capabilities.AutoRegisterEntityProvider;
+import org.sakaiproject.entitybroker.entityprovider.capabilities.Describeable;
+import org.sakaiproject.entitybroker.entityprovider.capabilities.Inputable;
+import org.sakaiproject.entitybroker.entityprovider.capabilities.Outputable;
+import org.sakaiproject.entitybroker.entityprovider.capabilities.PropertyProvideable;
+import org.sakaiproject.entitybroker.entityprovider.capabilities.Resolvable;
+import org.sakaiproject.entitybroker.entityprovider.capabilities.Sampleable;
+import org.sakaiproject.entitybroker.entityprovider.extension.Formats;
+import org.sakaiproject.entitybroker.exception.EntityException;
+import org.sakaiproject.entitybroker.exception.EntityNotFoundException;
+import org.sakaiproject.entitybroker.util.AbstractEntityProvider;
+import org.sakaiproject.exception.IdUnusedException;
+import org.sakaiproject.exception.PermissionException;
+import org.sakaiproject.mailarchive.api.MailArchiveService;
+import org.sakaiproject.message.api.Message;
+import org.sakaiproject.site.api.Site;
+import org.sakaiproject.site.api.SiteService;
+import org.sakaiproject.site.api.ToolConfiguration;
+import org.sakaiproject.time.api.Time;
+import org.sakaiproject.time.api.TimeService;
+import org.sakaiproject.tool.api.SessionManager;
+import org.sakaiproject.tool.api.ToolManager;
+import org.sakaiproject.user.api.UserDirectoryService;
+import org.sakaiproject.util.MergedList;
+import org.sakaiproject.util.ResourceLoader;
+import org.sakaiproject.util.Validator;
+
+/**
+ * Allows some basic functions on email archive. Due to limitations of
+ * EntityBroker the internal URLs of the mailArchiveMessages service can't be
+ * exposed directly, so we have to map them, with assumptions about characters
+ * used in IDs. Basically we pack together the
+ * {siteId}:{channelId}:{mailArchiveMessageId} into the ID.
+ *
+ */
+public class MailArchiveEntityProvider extends AbstractEntityProvider implements
+		EntityProvider, Outputable, Inputable, Describeable, ActionsExecutable {
+
+	public final static String ENTITY_PREFIX = "mailarchive";
+
+	private static final Log log = LogFactory
+			.getLog(MailArchiveEntityProvider.class);
+	private static ResourceLoader rb = new ResourceLoader("email");
+	public static int DEFAULT_NUM_MAILMESSAGES = 3;
+	public static int DEFAULT_DAYS_IN_PAST = 10;
+
+	/**
+	 * Prefix for this provider
+	 */
+	@Override
+	public String getEntityPrefix() {
+		return ENTITY_PREFIX;
+	}
+
+	// output formats
+	public String[] getHandledOutputFormats() {
+		return new String[] { Formats.JSON };
+	}
+
+	// input formats
+	public String[] getHandledInputFormats() {
```

Comment:
Not creating anything, this shouldn't be required. Might be implementing the wrong type or something too high up in the hierarchy.